hibernate-core/src/main/java/org/hibernate/dialect/Dialect.java

```diff
@@ -210,21 +210,24 @@ protected Dialect() {
 		// register hibernate types for default use in scalar sqlquery type auto detection
 		registerHibernateType( Types.BIGINT, StandardSpiBasicTypes.BIG_INTEGER.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.BINARY, StandardSpiBasicTypes.BINARY.getJavaTypeDescriptor().getTypeName() );
-		registerHibernateType( Types.BIT, StandardSpiBasicTypes.SHORT.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.BIT, 64, StandardSpiBasicTypes.LONG.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.BIT, 32, StandardSpiBasicTypes.INTEGER.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.BIT, 16, StandardSpiBasicTypes.SHORT.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.BIT, 8, StandardSpiBasicTypes.BYTE.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.BIT, 1, StandardSpiBasicTypes.BOOLEAN.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.BOOLEAN, StandardSpiBasicTypes.BOOLEAN.getJavaTypeDescriptor().getTypeName() );
-		registerHibernateType( Types.CHAR, StandardSpiBasicTypes.CHARACTER.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.CHAR, StandardSpiBasicTypes.STRING.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.CHAR, 1, StandardSpiBasicTypes.CHARACTER.getJavaTypeDescriptor().getTypeName() );
-		registerHibernateType( Types.CHAR, 255, StandardSpiBasicTypes.STRING.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.DATE, StandardSpiBasicTypes.DATE.getJavaTypeDescriptor().getTypeName() );
 		registerHibernateType( Types.DOUBLE, StandardSpiBasicTypes.DOUBLE.getJavaTypeDescriptor().getTypeName() );
-		registerHibernateType( Types.FLOAT, StandardSpiBasicTypes.FLOAT.getJavaTypeDescriptor().getTypeName() );
+		registerHibernateType( Types.FLOAT, StandardSpiBasicTypes.DOUBLE.getJavaTypeDescriptor().getTypeName() );
```

Comment:
I'm not 100% sure about this particular change right here.

Reply:
Well, I think it's right; the difference between `Types.FLOAT` and `Types.DOUBLE` has nothing to do with precision, so we shouldn't pretend it does.